### PR TITLE
feat(store): add array proxy

### PR DESCRIPTION
**Pull request by Saul-Mirone**

### CODE REVIEW

File: packages/blocks/src/database-block/database-model.ts [hunk 1]
```diff
@@ -180,9 +228,8 @@ export const DatabaseBlockSchema = defineBlockSchema({
   flavour: 'affine:database',
   props: (internal): Props => ({
     title: internal.Text(),
-    columns: [],
     yCells: internal.Map<Y.Map<unknown>>(),
```
Comment (doodlewind) [inline packages/blocks/src/database-block/database-model.ts:231]: Is it possible renaming `yCells` -> `cells`?

Comment (Saul-Mirone) [reply packages/blocks/src/database-block/database-model.ts:231]: Yeah, and we'll have an object proxy for it.

**Review by Himself65 — COMMENTED** [review]
unit test case?

**Comment by Saul-Mirone** [comment]
> Member
> ###

Done

**Comment by Saul-Mirone** [comment]
> unit test case?

Done

**Review by doodlewind — APPROVED** [review]

**Pull request merged**